infra/discovery/src/listener/listener.js

```diff
@@ -134,84 +136,92 @@ async function main() {
   do {
     start = new Date()
 
-    // Compute the range of blocks to process,
-    // while respecting trailing block configuration.
+    // Compute the range of blocks to process
     const currentBlock = await withRetrys(async () => {
       return context.web3.eth.getBlockNumber()
     })
-    const toBlock = Math.max(currentBlock - context.config.trailBlocks, 0)
+    const toBlock = Math.max(currentBlock, 0)
 
-    // Listener is up to date. Nothing to do.
-    if (toBlock <= processedToBlock) {
-      logger.debug('No new blocks to process')
-      continue
-    }
-    logger.info(
-      `Querying events within interval (${processedToBlock}, ${toBlock}]`
-    )
+    for (const contractKey of Object.keys(contracts)) {
+      let processedToBlock = await getLastBlock(
+        context.config,
+        `${contractKey}_`
+      )
+      logger.debug(`Processing from ${processedToBlock} for ${contractKey}`)
+
+      // Listener is up to date. Nothing to do.
+      if (toBlock <= processedToBlock) {
+        logger.debug('No new blocks to process')
+        continue
+      }
+      logger.info(
+        `Querying events within interval (${processedToBlock}, ${toBlock})`
+      )
 
-    // Retrieve all events for the relevant contracts
-    const eventArrays = await Promise.all([
-      withRetrys(async () => {
-        return contractsContext.marketplace.eventCache.allEvents()
-      }),
-      withRetrys(async () => {
-        return contractsContext.identityEvents.eventCache.allEvents()
+      // Retrieve all events for the relevant contracts
+      const eventArrays = await withRetrys(async () => {
+        return contracts[contractKey].eventCache.allEvents()
       })
-    ])
 
-    // Flatten array of arrays filtering out anything undefined
-    const events = [].concat(...eventArrays.filter(x => x))
+      // Flatten array of arrays filtering out anything undefined and filter
+      // out events outside of interval (processedToBlock, toBlock].
+      const newEvents = [].concat(
+        ...eventArrays.filter(event => {
+          if (
+            event &&
+            event.blockNumber > processedToBlock &&
+            event.blockNumber <= toBlock
```

Comment:
do we still need this filtering on toBlock ? wouldn't it be ok to process all events > processedToBlock ?

Reply:
Honestly, I'm not really sure why this check was there to begin with?  I tend to try not to fiddle with defensive checks added by people before me, just in case they knew something I didn't.

Reply:
Do you mind removing it to clean up as much as possible this logic ?